sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala

```diff
@@ -4094,9 +4100,16 @@ case class ArrayExcept(left: Expression, right: Expression) extends ArrayBinaryL
             }
           } else {
             val elem = array1.get(i, elementType)
-            if (!hs.contains(elem)) {
-              arrayBuffer += elem
-              hs.add(elem)
+            if (isNaN(elem)) {
+              if (notFoundNaNElement) {
+                arrayBuffer += elem
```

Comment:
For this, let's wait a little bit for the decision at the first PR.
- https://github.com/apache/spark/pull/33955/files#r708570515

Reply:
Done